language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaLanguageServer.java

```diff
@@ -89,9 +90,10 @@ public class BallerinaLanguageServer extends AbstractExtendedLanguageServer
     private ExtendedLanguageClient client = null;
     private final TextDocumentService textService;
     private final WorkspaceService workspaceService;
+
+    private LSClientCapabilities capabilities;
```

Comment:
This is incorrect. We hold everything inside the `LanguageServerContext`. No need to keep anything at instance level unless necessary. We can simply to ` LSClientCapabilities clientCapabilities = serverContext.get(LSClientCapabilities.class);` to get the client capabilities from LS context.